src/Config.ts

```diff
@@ -185,9 +189,17 @@ export interface IConfigDatastore {
 
 export interface IConfigRoomRule {
     /**
-     * Room ID or alias
+     * Room ID or alias. Optional.
+     *
+     * Either `room` or `roomRegex` must be provided.
```

Comment:
Have you considered splitting this into three types?
`type IConfigRoomRule = IConfigRoomRuleWithRoom | IConfigRoomRuleWithRoomRegex;`

There would need to be one validation when loading the config, but after that it reduces the complexity of conditions.